src/workers/scran.worker.js

```diff
@@ -21,694 +21,685 @@ let preflights_summary = {};
-            kana_db
-                .loadAnalysis(id)
-                .then(async (res) => {
-                    if (res == null) {
-                        postMessage({
-                            type: "KanaDB_ERROR",
-                            msg: `Fail: cannot load analysis ID '${id}'`,
-                        });
-                    } else {
-                        let params = await unserializeAllSteps(res);
-
-                        var transferrable = [];
-                        extractBuffers(params.other, transferrable);
-                        postMessage(
-                            {
-                                type: "loadedParameters",
-                                resp: params,
-                            },
-                            transferrable
-                        );
-                    }
-                })
-                .catch((err) => {
-                    console.error(err);
-                    postError(type, err, fatal);
-                });
+        postMessage({
+          type: "PREFLIGHT_INPUT_DATA",
+          resp: resp,
+          msg: "Success: PREFLIGHT_INPUT done",
+        });
+      })
+      .catch((err) => {
+        console.error(err);
+        postError(type, err, fatal);
+      });
+
+    /**************** VERSUS MODE *******************/
+  } else if (type == "computeVersusClusters") {
+    loaded
+      .then((x) => {
+        let rank_type = payload.rank_type.replace(/-.*/, ""); // summary type doesn't matter for pairwise comparisons.
+        let res = superstate.marker_detection.computeVersus(
+          payload.left,
+          payload.right,
+          rank_type,
+          payload.modality
+        );
+        postMessage({
+          type: "computeVersusClusters",
+          resp: res,
+          msg: "Success: COMPUTE_VERSUS_CLUSTERS done",
+        });
+      })
+      .catch((err) => {
+        console.error(err);
+        postError(type, err, fatal);
+      });
+  } else if (type == "computeVersusSelections") {
+    loaded
+      .then((x) => {
+        let rank_type = payload.rank_type.replace(/-.*/, ""); // summary type doesn't matter for pairwise comparisons.
+        let res = superstate.custom_selections.computeVersus(
+          payload.left,
+          payload.right,
+          rank_type,
+          payload.modality
+        );
+        postMessage({
+          type: "computeVersusSelections",
+          resp: res,
+          msg: "Success: COMPUTE_VERSUS_SELECTIONS done",
+        });
+      })
+      .catch((err) => {
+        console.error(err);
+        postError(type, err, fatal);
+      });
+
+    /**************** OTHER EVENTS FROM UI *******************/
+  } else if (type == "getMarkersForCluster") {
+    loaded
+      .then((x) => {
+        let cluster = payload.cluster;
+        let rank_type = payload.rank_type;
+        let modality = payload.modality;
+        var resp = superstate.marker_detection.fetchGroupoutputs(
```

Comment:
This part is now something like:

```
var raw_res = superstate.marker_detection.fetchResults()[modality];
let resp = formatMarkerResults(raw_res, cluster, rank_type);
```

where `formatMarkerResults` is taken from https://github.com/kanaverse/bakana/blob/1.2.0/src/steps/utils/markers.js#L74-L139.